### ISSUE TITLE
Run 1 input file transfer

### ISSUE THREAD

**Issue by drphilmarshall** [DarkEnergyScienceCollaboration/Twinkles#133]
@rbiswas4 @jbkalmbach can you please move your PhoSim input files to a web-visible directory and share the URL? @TomGlanzman can then pull them down with a wget script and start running jobs. Thanks! :-)

For reference:

* Each visit's inputs amount to not more than 100Mb of data, consisting of a gzipped plain text instance catalog and a corresponding gzipped tarball of SED files of comparable size. 
* Each input takes about 10 minutes to make
* At the time of writing, about 100 inputs have been made, meaning that we are looking at about 10Gb to transfer initially. 
* The full Run 1 set will contain about 1600 inputs (in all filters), or at most 160 Gb of data. The inputs are being made (almost) sequentially, starting with seasons 2 and 3. A fallback Run 1 dataset would be to only make seasons 2, 3 and 4, for about 500 visits.

**Comment by cwwalter** [comment]
You might consider using.

http://www.nersc.gov/users/storage-and-file-systems/transferring-data/globus-online/

NERSC is and I suspect SLAC will be too because of ATLAS.

especially if Washington is an endpoint.

**Comment by rbiswas4** [comment]
yes, globus is quite good.

Rahul

**Comment by TomGlanzman** [comment]
We can try globus - I routinely use the CLI to transfer data from BNL to SLAC.  Have not tried the web interface, but if your data files are in a globus-visible, give me the coordinates of a few examples and I'll give it a try.

   - Tom

**Comment by cwwalter** [comment]
I just tried it with the web interface.  Super easy and slick.  I didn't even need to make a globus account.  It let me log in with my Duke Credentials.

There seem to be a bunch of SLAC endpoints:

slac#lsstbnl
slac#lcls
slac#ssrl
slac#ztest
slac#osg
slac#kipac
wdeng#SLACXRD

Then I did a one click thing and it turned my laptop into an endpoint too.

I guess the first one is the one you use. Don't see anything too obvious for UW though.  Maybe Rahul knows.

**Comment by rbiswas4** [comment]
OK ... I have note explored this yet, Just trying to get a situation where we can get started. I have managed to put a handful of files up on a website, and put in this script https://github.com/DarkEnergyScienceCollaboration/Twinkles/blob/master/transfers/downloadInstanceCatalogs.sh
which should download all the files that you need.

**Comment by TomGlanzman** [comment]
Thanks Rahul, got them (19 instance catalogs).

   - Tom

**Comment by drphilmarshall** [comment]
Excellent! Thanks @rbiswas4 . Tom, did you get the gzipped tarballs of supernova SEDs as well?

**Comment by TomGlanzman** [comment]
No, the only files were gzipped instance catalogs.  @rbiswas4 can you direct me to the SED files?  Also, can you provide some detail on how they might be used, i.e., can they all sit along side the existing SED files?  Or how would you recommend they be invoked by phoSim?

**Comment by rbiswas4** [comment]
You should be getting two sets of files. basename.txt.gz and basename.tar.gz. The first is a phosim instance catalog, and the second is a tarball of the spectra associated with this instance catalog.

**Comment by johnrpeterson** [comment]
Globus is where the phosim release data goes, by the way.  Sent this message a while ago:



Hi all-

It took us a while to set this up, but we now have a pretty nice way of transfering large quantities of phosim simulations back and forth.  En-Hsin, Glenn, and research computing at Purdue set up this large storage end point of our large scale phosim runs on OSG/Diagrid with this “globus” file sharing service.

Some of you may know more than me, but it seems pretty useful to us so far.  The new interface works somewhat like “dropbox”, but for serious quantities of data (>Tbytes).

All you have to do is go to  www.globus.org<http://www.globus.org> and set up an account.  Then you can make anything an endpoint (like your laptop).

If you want to see the phosim files go to the “endpoint":

purduelsst#phosim

and then look in

/depot/lsst/phosim/data/v3.4

Here we have all the calibration data (flats, biases, darks), and some focal planes of stars & galaxies all run with phosim v3.4.2.   You can download either small numbers of files or all of it (~0.5 Tb) relatively quickly. This way no one has to set up accounts for anyone, it all can be shared freely and efficiently, and should be scalable.

I hope this is useful, and for the moment will be hosting the general-purpose phosim outputs there.

Thanks,

John


________________________________

Use REPLY-ALL to reply to list

To unsubscribe from the LSST-DESC-SURVEYSIM list, click the following link:
https://listserv.slac.stanford.edu/cgi-bin/wa?SUBED1=LSST-DESC-SURVEYSIM&A=1

On Feb 18, 2016, at 5:02 PM, Chris Walter <notifications@github.com<mailto:notifications@github.com>> wrote:


I just tried it with the web interface. Super easy and slick. I didn't even need to make a globus account. It let me log in with my Duke Credentials.

There seem to be a bunch of SLAC endpoints:

slac#lsstbnl
slac#lcls
slac#ssrl
slac#ztest
slac#osg
slac#kipac
wdeng#SLACXRD

Then I did a one click thing and it turned my laptop into an endpoint too.

I guess the first one is the one you use. Don't see anything too obvious for UW though. Maybe Rahul knows.

**Comment by drphilmarshall** [comment]
Thanks all - Run 1 files are now transferred (or are on the way), so I'll close this out - but I'll issue the globus investigation for Twinkles 1.

**Comment by rbiswas4** [comment]
Sorry to reopen the issue: but it is for good news:
I ran a check and made sure that everything has been transferred to nersc. There should be 1227 pairs of phosim instance catalogs and spectra for these. I have run two checks on these:
- I checked that all SN spectra files alluded to in the phosim instance catalog are indeed in the tar file
- that each of the tar files are not empty and have a substantial data storage. 
Further validation has not been done, but this will hopefully take care of the most basic mistakes. 

Now, I know that I had given a number of 1680 at some point. This was simply my mistake of muddling different numbers. I went back and checked and the plan we had decided on has always had  1227 visits (as in the csv files we have shared). I can only guess where the 1680 number came from (I had done a calculation of adding a bunch of observations over a few month period in addition to the 10 year plan, just to see how the additional cadence of the DDF stacks up, and had been playing with the idea that we could do a few more this way. the 1680 probably came from that calculation.
